src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -22,19 +39,24 @@ export const TodoFilter: React.FC = () => {
           type="text"
           className="input"
           placeholder="Search..."
+          value={query}
+          onChange={(event) => handleQuery(event.target.value.trim())}
         />
         <span className="icon is-left">
           <i className="fas fa-magnifying-glass" />
         </span>
 
-        <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-          <button
-            data-cy="clearSearchButton"
-            type="button"
-            className="delete"
-          />
-        </span>
+        {query.trim().length > 0 && (
```

Comment:
I see you like it so much)
so length could be 0,1,2,3,4... and so on. 
There are falsy values which are interprented as false in case you use it in conditions or just transforming into `Boolean()`. It' values like `0`, emply sting (`''`), `null`. 

So you can just write `query.trim().length` and in case it's `0` - you'll get `false` value, in all other cases (1,2,3...) it will be `true`
```suggestion
        {query.trim().length && (
```